pallets/moonbeam-lazy-migrations/src/lib.rs

```diff
@@ -124,5 +130,55 @@ pub mod pallet {
 			}
 			Ok(Pays::No.into())
 		}
+		#[pallet::call_index(2)]
+		#[pallet::weight(Pallet::<T>::create_contract_metadata_weight(MAX_CONTRACT_CODE_SIZE))]
+		pub fn create_contract_metadata(
+			origin: OriginFor<T>,
+			address: H160,
+		) -> DispatchResultWithPostInfo {
+			ensure_signed(origin)?;
+
+			ensure!(
+				pallet_evm::AccountCodesMetadata::<T>::get(address).is_none(),
+				Error::<T>::ContractMetadataAlreadySet
+			);
+
+			// Ensure contract exist
+			let code = pallet_evm::AccountCodes::<T>::get(address);
+			ensure!(!code.is_empty(), Error::<T>::ContractNotExist);
+
+			// Construct metadata
+			let code_size = code.len() as u64;
+			let code_hash = sp_core::H256::from(sp_io::hashing::keccak_256(&code));
+			let meta = pallet_evm::CodeMetadata {
+				size: code_size,
+				hash: code_hash,
+			};
+
+			// Set metadata
+			pallet_evm::AccountCodesMetadata::<T>::insert(address, meta);
+
+			Ok((
+				Some(Self::create_contract_metadata_weight(code_size)),
+				Pays::No,
+			)
+				.into())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		fn create_contract_metadata_weight(code_size: u64) -> Weight {
+			// max entry size of AccountCodesMetadata (full key + value)
+			const PROOF_SIZE_CODE_METADATA: u64 = 100;
+			// intermediates nodes might be up to 3Kb
+			const PROOF_SIZE_INTERMEDIATES_NODES: u64 = 3 * 1024;
+
+			// Account for 2 reads, 1 write
+			<T as frame_system::Config>::DbWeight::get()
+				.reads_writes(2, 1)
+				.set_proof_size(
+					code_size + (PROOF_SIZE_INTERMEDIATES_NODES * 2) + PROOF_SIZE_CODE_METADATA,
```

Comment:
```suggestion
			const PROOF_SIZE_INTERMEDIATES_NODES: u64 = 6 * 1024;

			// Account for 2 reads, 1 write
			<T as frame_system::Config>::DbWeight::get()
				.reads_writes(2, 1)
				.set_proof_size(
					code_size + PROOF_SIZE_INTERMEDIATES_NODES + PROOF_SIZE_CODE_METADATA,
```

Reply:
Times two is expected, we touch 2 keys so we should account 2 times the intermediates nodes